src/common/pool_map.c

```diff
@@ -1347,14 +1347,17 @@ pool_map_merge(struct pool_map *map, uint32_t version,
 			/* new buffer may have changes for this domain */
 			if (sdom->do_children != NULL) {
 				struct pool_domain *child = addr;
+				struct pool_comp_cntr	s_cntr;
 
-				D_DEBUG(DB_TRACE, "Scan children of %s[%d]\n",
+				pool_tree_count(sdom, &s_cntr);
+				D_DEBUG(DB_TRACE, "Scan children of %s[%d] tgt_nr %u\n",
 					pool_domain_name(ddom),
-					ddom->do_comp.co_id);
+					ddom->do_comp.co_id, s_cntr.cc_targets);
 
 				if (ddom->do_children == NULL)
 					ddom->do_children = child;
 
+				ddom->do_target_nr += s_cntr.cc_targets;
```

Comment:
My understanding is that this should never happen, since this function is only used for extending, and these targets are from the extended target list anyway.

But this pool map merge code are very confusing to me.

Reply:
ok, but for the code logic, seems better to do like my suggest.
But I am fine to leave it as now.